src/layouts/enum.py

```diff
@@ -0,0 +1,43 @@
+from typing import Dict
+
+
+class Enum:
+    """Mapping of str values to uint32 integer."""
+
+    _MAX_VAL = 4294967295  # 2^32 - 1
+
+    def __init__(self, values: Dict[str, int]):
```

Comment:
I am thinking if it will be a better interface if we can just do `Enum(["buy", "sell"])`. And the integer will just be the index.

Reply:
That way we also don't need to have the validation logic.